src/go/k8s/controllers/redpanda/redpanda_node_pvc_controller.go

```diff
@@ -0,0 +1,186 @@
+	if result.RequeueAfter > 0 {
+		durationMsg = fmt.Sprintf("%s, next run in %s", durationMsg, result.RequeueAfter.String())
+	}
+	log.Info(durationMsg)
+
+	return result, err
+}
+
+// nolint:funlen,unparam // the length is ok
+func (r *RedpandaNodePVCReconciler) reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx).WithName("RedpandaNodePVCReconciler.reconcile")
+	Infof(log, "detected node %s deleted; checking if any PVC should be removed", req.Name)
+
+	redpandaNameList := make([]string, 0)
+	if r.OperatorMode {
+		opts := &client.ListOptions{Namespace: req.Namespace}
+		redpandaList := &v1alpha1.RedpandaList{}
+		if errList := r.Client.List(ctx, redpandaList, opts); errList != nil {
+			return ctrl.Result{}, fmt.Errorf("could not GET list of Redpandas in namespace: %w", errList)
+		}
+
+		for i := range redpandaList.Items {
+			item := redpandaList.Items[i]
+			redpandaNameList = append(redpandaNameList, item.Name)
+		}
+	} else {
+		releaseName, ok := os.LookupEnv(EnvHelmReleaseNameKey)
+		if !ok {
+			Infof(log, "Skipping reconciliation, expected to find release-name env var: %q", EnvHelmReleaseNameKey)
+			return ctrl.Result{}, nil
+		}
+
+		redpandaNameList = append(redpandaNameList, releaseName)
+	}
+
+	opts := &client.ListOptions{Namespace: req.Namespace}
+	pvcList := &corev1.PersistentVolumeClaimList{}
+	if errGET := r.Client.List(ctx, pvcList, opts); errGET != nil {
+		return ctrl.Result{}, fmt.Errorf("could not GET list of PVCs: %w", errGET)
+	}
+
+	var errs error
+	// creating map of release name to sideCarEnabled
+	doTaskMap := make(map[string]bool, 0)
+	// this could be a lot of redpandas, usually 1 but may be a few
+	for i := range pvcList.Items {
+		pvc := pvcList.Items[i]
+
+		// first check if the application label is redpanda, if not continue
+		if len(pvc.Labels) == 0 {
+			continue
+		}
+
+		if _, ok := pvc.Labels[K8sNameLabelKey]; !ok {
+			continue
+		}
+
+		if len(pvc.Annotations) == 0 {
+			continue
+		}
+
+		// Now we check if the node where the PVC was originally located was deleted
+		// if it is, then we change the PV to change the reclaim policy and then mark the pvc for deletion
+		workerNode, ok := pvc.Annotations[volume.AnnSelectedNode]
+
+		if !ok || workerNode == "" {
+			errs = errors.Join(errs, fmt.Errorf("worker node annotation not found or node name is empty: %q", workerNode)) //nolint:goerr113 // joining since we do not error here
+		}
+
+		// we are not being deleted, move on
+		if workerNode != req.Name {
+			continue
+		}
+
+		// Now check if we are allowed to operate on this pvc as there may be another controller working
+		var releaseName string
+		if val, ok := pvc.Labels[K8sInstanceLabelKey]; !ok || !isValidReleaseName(val, redpandaNameList) {
+			Infof(log, "could not find instance label or unable retrieve valid releaseName: %s", val)
+			return ctrl.Result{}, nil
```

Comment:
I was under impression that we log any error, but will not stop processing all PVC. Why we returning at this particular moment?